### PR TITLE
Feat: add paket parsing

### PR DESCRIPTION
**Pull request by orsagie**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Refactor nuget parsing and add `paket` support in the plugin

### CODE REVIEW

File: lib/dotnet-parser.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
+var fs = require('fs');
```
Comment (lili2311) [inline lib/dotnet-parser.js]: can we use `const` and `let` instead?

Comment (orsagie) [reply lib/dotnet-parser.js]: yeah, just noticed. changing.

Comment (orsagie) [reply lib/dotnet-parser.js]: I thought you wrote - `we can't use const and let` - which is true because of node 4

Comment (Kirill89) [reply lib/dotnet-parser.js]: We need to refactor this. Maybe to `ts`. But I believe not as part of this PR.

**Pull request merged**